### PR TITLE
Prisma2

### PR DESCRIPTION
**Pull request by andriisoldatenko**
we need change houston-backend-secret to add `?schema=houston$default` to DATABASE_URL for prisma
to be able to run it against existing tables.

**after this change:**
```
k get secrets astronomer-1593696217-houston-backend -n astronomer -o=jsonpath='{.data.connection}' | base64 -D
postgres://******:*******@astronomer-1593696217-postgresql.astronomer.svc.cluster.local:5432/astronomer_1593696217_houston?schema=houston%24default
```

Links:
- https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema#example

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -100,19 +99,31 @@ def patch_conn_secret(kube, namespace, secret_name, connection):
...
 
+    if db_schema is not None:
+        string_data = {'connection': '{conn}?schema={schema}'.format(
+            conn=conn,
+            # we need to escape schema to URL format e.g. houston$default -> houston$default
+            # safe='%' prevent to double escape if schema already good: 'houston%24default'
+            # In [32]: quote(quote('houston$default'), safe='%')
+            # Out[32]: 'houston%24default'
+            schema=quote(db_schema, safe='%')
+        )}
```
Comment (sjmiller609) [inline main.py:115]: What happens if this is run multiple times? I'm not sure, but it kind of looks to me like this fetches the DB string, then mutates it, I worry a second run would append the schema part redundantly. Can we add some unit testing around this condition, please?

File: main.py [hunk 2]
```diff
@@ -42,9 +42,11 @@ def ensure_db(conn):
 
     # database didn't exist with `-` or `_`, so create it with `_`
     click.echo("Database does not exist, creating...")
```
Comment (adam2k) [inline main.py:44]:
```suggestion
   click.echo("Attempting to create database...")
```

File: main.py (hunk 1 above)
Comment (andriisoldatenko) [reply main.py]:
```
# In [32]: quote(quote('houston$default'), safe='%')
# Out[32]: 'houston%24default'
```

we don't have test framework or CI/CD for tests in this repo.

**Review by sjmiller609 — CHANGES_REQUESTED** [review]
I think we need tests to ensure that the idempotency is working. I can help get that set up

File: test/component/test_db_bootstrapper.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    sleep(5)
+    return postgres.short_id
+
+def test_db_bootstrapper_user(db_bootstrapper):
+    assert "root" == db_bootstrapper.check_output("whoami")
+
+def test_kubectl_is_working(db_bootstrapper):
+    assert "kind" in db_bootstrapper.check_output("kubectl get nodes")
+
+def test_db_bootstrapper_idempotent(db_bootstrapper, kube_client):
```
Comment (sjmiller609) [inline test/component/test_db_bootstrapper.py:58]: this is the behavior I was mostly worried about

**Review by sjmiller609 — APPROVED** [review]
I added a test for the item of concern. If tests pass, I am good with this change.

File: .circleci/config.yml [hunk 4]
```diff
@@ -29,6 +29,12 @@ jobs:
       - push:
           comma_separated_tags: "$CIRCLE_BRANCH"
           image_name: ap-db-bootstrapper
+  component-test:
+    machine:
+      image: ubuntu-1604:201903-01
+      resource_class: xlarge
```
Comment (adam2k) [inline .circleci/config.yml:35]: Do we need this particular image or is there more lightweight version of linux, like alpine, that we can use that is closer to what our staging and prod envs are using?

File: main.py [hunk 5]
```diff
@@ -23,6 +36,19 @@ def get_new_db(engine, db_name):
     return conn
 
 
+def ensure_schema(conn, db_schema):
+    """
+    Create the PostgreSQL schema if it does not exist.
+    """
+    engine = create_engine(conn, isolation_level='AUTOCOMMIT')
+    logging.debug(f"Using connection string: {conn}")
```
Comment (kushalmalani) [inline main.py:44]: Is there a typo in the debug message? f"Using

File: main.py [hunk 6]
```diff
@@ -72,70 +100,97 @@ def create_conn_secret(kube, namespace, secret_name, connection):
         api_version="v1",
         kind="Secret",
         metadata=metadata,
-        string_data={'connection': connection})
+        string_data=string_data)
 
-    try:
-        kube.create_namespaced_secret(namespace, body)
-        click.echo("Successfully created secret")
-    except Exception as e:
```
Comment (kushalmalani) [inline main.py:80]: Why are we removing this? Does it not ever go in except block?

**Review by sjmiller609 — APPROVED** [review]

File: test/component/test_db_bootstrapper.py (hunk 3 above)
Comment (adam2k) [reply test/component/test_db_bootstrapper.py:84]: @sjmiller609 does this test cover everything that you were worried about?

**Review by adam2k — COMMENTED** [review]
This looks good to me.  I just have the single question about using the Ubuntu image instead of Alpine.

File: main.py (hunk 5 above)
Comment (andriisoldatenko) [reply main.py:44]: nope it's python f-strings https://www.python.org/dev/peps/pep-0498/

File: .circleci/config.yml (hunk 4 above)
Comment (andriisoldatenko) [reply .circleci/config.yml:35]: @sjmiller609 using it for all e2e tests AFAIK.

File: main.py (hunk 6 above)
Comment (andriisoldatenko) [reply main.py:80]: @sjmiller609 suggested to see error instead if hiding it from logs.

File: test/component/test_db_bootstrapper.py (hunk 3 above)
Comment (andriisoldatenko) [reply test/component/test_db_bootstrapper.py:84]: yep.

**Review by kushalmalani — APPROVED** [review]

**Review by ashb — COMMENTED** [review]
LGTM from a very quick look.

**Pull request merged**